src/web/extractors.rs

```diff
@@ -1046,7 +1062,11 @@ pub struct BsoQueryParams {
     pub limit: Option<u32>,
 
     /// position at which to restart search (string)
-    pub offset: Option<u64>,
+    #[serde(deserialize_with = "deserialize_offset")]
+    pub offset: Option<Offset>,
+
+    pub newer_eq: Option<i64>,
```

Comment:
I don't think newer/older_eq belong here, they aren't ever populated. Looks their values are determined when looking at the sort type: https://github.com/mozilla-services/server-syncstorage/blob/af652cc/syncstorage/storage/sql/__init__.py#L480

Reply:
Also related to the link above: there should be some kind of check on this timestamp value vs the `older/newer` params (see the `raise InvalidOffsetError` cases)

Reply:
It's pretty weird how the python code uses the params dictionary as a bag of state and puts newer_eq in it when it will never be in a url. But yeah I think these can be removed.

Yes, I think the missing raise InvalidOffsetError equivalent is what is causing the one integration test to fail.

Reply:
I don't think I fully understand this optimization yet, but another piece from the Python that strikes me as maybe odd is it seems like it can create these `newer/older_eq` params when there may also be an `newer/older` as well?

Which would produce 2 comparisons on modified like `modified > value1 AND modified >= value2`.